source/agora/test/ManyValidators.d

```diff
@@ -75,6 +80,10 @@ void manyValidators (size_t validators)
             format("Node %s has validator count %s. Expected: %s",
                 idx, node.countActive(Height(GenesisValidatorCycle + 1)), validators)));
 
+    // Wait for nodes to run a discovery task and update their required peers
+    Thread.sleep(3.seconds);
+    network.waitForDiscovery();
```

Comment:
Shouldn't `waitForDiscovery` handle the sleeping period ?

Reply:
Even after block externalization, required_peers in NetworkManager is not updated until the next call to `discover()` which happens periodically in discovery timer. So between the block externalization and the next timer tick, NetworkManager still thinks the network state is complete, even though quorums were shuffled in the last block and makes `waitForDiscovery()` return earlier than it should.

Reply:
So.... Shouldn't we fix this ?